grafana/grafana
# InfluxDB: Flux: handle boolean data-columns

fixes https://github.com/grafana/grafana/issues/34296

In InfluxDB Flux mode, data can contain booleans, these were not handled properly in grafana. we assumed boolean data has a column-type "bool", but it is "boolean". This pull request fixes it.
 i changed the list of data-types to be based on the influxdb documentation.
i updated the tests:
- there was a test already with datafile named "boolean.csv", but it dealt with boolean tags (labels). so i renamed it to "boolean_tags.csv", and added a new testfile named "boolean_data.csv"

how to test:
- use `make devenv sources=influxdb`
- unfortunately we do not have booleans in that database by default, so we write a flux query which looks at a numeric value in the database, and returns true if it is above 99.2 or false if it is below:
```
from(bucket: "mybucket")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "cpu")
  |> filter(fn: (r) => r["_field"] == "usage_idle")
  |> filter(fn: (r) => r["cpu"] == "cpu1")
  |> map(fn: (r) => ({
    _time: r._time,
    _value: if r._value > 99.2 then true else false
  }))
```
- go to explore-mode, choose the `gdev-influxdb-influxql` mode, and paste it in and run it. you should see the timestamps and booleans in the table-view. (on `main` branch at this point you would get an error saying `no matching converter found for [boolean]`)
